### PR TITLE
Syscollector test module refactor

### PR DESCRIPTION
**Pull request by BelenValdivia**
|Related issue|
|-------------|
|   https://github.com/wazuh/wazuh-qa/issues/3541    |

## Description

After the last restructure of VD, we propose that this new structure will be applied to the Syscollector suite of our IT.

### Added

- New structure to Syscollector test:

```
integration/
└ test_analisysd/
      └ test_syscollector/
	 │
	 ├ data/
         │   └ rules/
	 │   │ │
	 │   │ └ syscollector_rules.xml
         │   │
	 │   └ test_cases/
	 │       │
         │       └ cases_syscollector_integration.yaml
         │ 
         └ test_syscollector_integration.py

```

---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @BelenValdivia (Developer)  |  test_syscollector/ | [🟢](https://ci.wazuh.info/job/Test_integration/33790/)  [🟢](https://ci.wazuh.info/job/Test_integration/33792/)  [🟢](https://ci.wazuh.info/job/Test_integration/33793/) | [🟢](https://github.com/wazuh/wazuh-qa/files/9983489/3541-R1-refactor-belenvaldivia.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9983490/3541-R2-refactor-belenvaldivia.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9983492/3541-R3-refactor-belenvaldivia.zi)  |   Ubuntu      |  5472b3a7a37201b6fe6c67e78faa6b872017889c      | Nothing to highlight |
| @QU3B1M  (Developer)   |    `test_analysisd/test_syscollector/`        | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39681) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39682) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39683)  |  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/11559354/report.zip)[:green_circle: ](https://github.com/wazuh/wazuh-qa/files/11559355/report1.zip)[:green_circle: ](https://github.com/wazuh/wazuh-qa/files/11559356/report2.zip)        |   Ubuntu     |    008183e5deff23cbd48756d2c63e1a91cb4c9dbc     | Execution with new changes |

### CODE REVIEW

File: tests/integration/test_analysisd/test_syscollector/test_syscollector_integration.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+from wazuh_testing.tools import (ANALYSISD_QUEUE_SOCKET_PATH, ALERT_FILE_PATH)
+from wazuh_testing.analysis import CallbackWithContext, callback_check_syscollector_alert
+
+pytestmark = [pytest.mark.server]
+
+# Generic vars
+TEST_DATA_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+TEST_CASES_PATH = os.path.join(TEST_DATA_PATH, 'test_cases')
+TEST_RULES_PATH = os.path.join(TEST_DATA_PATH, 'rules')
+local_internal_options = {'analysisd.debug': '2'}
```
Comment (damarisg) [inline tests/integration/test_analysisd/test_syscollector/test_syscollector_integration.py:14]: Why didn't has defined these values on the `wazuh_testing/modules/analisisd/__init__.py` file?

**Review by damarisg — CHANGES_REQUESTED** [review]
This PR is for `4.4.0,` but it has `4.5.0`. Could change it, please?

File: tests/integration/test_analysisd/test_syscollector/test_syscollector_integration.py (hunk 1 above)
Comment (QU3B1M) [reply tests/integration/test_analysisd/test_syscollector/test_syscollector_integration.py:14]: Those are values for the test scenario, if we need to move them we could send them to the __init__.py from this test module

File: tests/integration/test_analysisd/test_syscollector/test_syscollector_integration.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+os_version:
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - Ubuntu Focal
+    - Ubuntu Bionic
+references:
+    - https://documentation.wazuh.com/current/user-manual/capabilities/syscollector.html\
+        #using-syscollector-information-to-trigger-alerts
```
Comment (damarisg) [inline tests/integration/test_analysisd/test_syscollector/test_syscollector_integration.py:26]: Please, Add spaces.

File: tests/integration/test_analysisd/test_syscollector/test_syscollector_integration.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    agent_id = mock_agent_module
+    event_header = f"d:[{agent_id}] {metadata['event_header']}"
+
+    # Add agent_id alert check
+    alert_expected_values = metadata['alert_expected_values']
+    alert_expected_values['agent.id'] = agent_id
+
+    # Create full message by header and payload concatenation
+    test_msg = event_header + metadata['event_payload']
+    print(test_msg)
```
Comment (damarisg) [inline tests/integration/test_analysisd/test_syscollector/test_syscollector_integration.py:118]: Please, remove it.

**Review by damarisg — CHANGES_REQUESTED** [review]

File: tests/integration/test_analysisd/test_syscollector/test_syscollector_integration.py (hunk 3 above)
Comment (QU3B1M) [reply tests/integration/test_analysisd/test_syscollector/test_syscollector_integration.py]: Removed on commit 008183e5deff23cbd48756d2c63e1a91cb4c9dbc

**Review by damarisg — APPROVED** [review]

**Pull request merged**